### PR TITLE
Bug 1304410 - Run ipynb using nbconvert

### PR DESCRIPTION
**Pull request by maurodoglio**
Using pyspark + jupyter nbconvert to execute notebook seems to be more
standard and it's gonna be a safer choice for future versions of spark.
Plus, we don't depend on runipy anymore!

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-airflow/45)

### CODE REVIEW

File: ansible/files/spark/airflow.sh [hunk 1]
```diff
@@ -81,20 +81,26 @@ fi
...
 
 if [[ $uri == *.jar ]]; then
     time env $environment spark-submit $runner_args --master yarn-client "./$job" $args
 elif [[ $uri == *.ipynb ]]; then
-    time env $environment runipy $runner_args "./$job" "./output/$job" --pylab
+    time env $environment \
+    PYSPARK_DRIVER_PYTHON=jupyter \
+    PYSPARK_DRIVER_PYTHON_OPTS="nbconvert --to notebook --log-level=10 --execute ./${job} --allow-errors --output ./output/${job}" \
+    pyspark
+    NOTEBOOK_EXIT_CODE=$?
```
Comment (vitillo) [inline ansible/files/spark/airflow.sh]: Could you please write a comment on why this is needed for future reference?

File: ansible/files/spark/airflow.sh [hunk 2]
```diff
@@ -81,20 +81,26 @@ fi
 # Run job
 job="${uri##*/}"
 cd $wd
+NOTEBOOK_EXIT_CODE=0
```
Comment (vitillo) [inline ansible/files/spark/airflow.sh]: Please use lower case variables names

File: ansible/files/spark/airflow.sh (hunk 1 above)
Comment (vitillo) [inline ansible/files/spark/airflow.sh]: What about using `rc` and do the assignment in each branch?

File: ansible/files/spark/airflow.sh [hunk 3]
```diff
@@ -112,3 +118,9 @@ do
 
     eval $upload_cmd
 done
+
+if [[ $rc != 0 ]]; then
+    exit $rc;
+elif [[ $NOTEBOOK_EXIT_CODE != 0 ]]; then
```
Comment (vitillo) [inline ansible/files/spark/airflow.sh]: If we use only `rc` we can remove this part.

**Comment by maurodoglio** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[ansible/files/spark/airflow.sh, line 84 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-airflow/45#-KSKoMvhXN7q7MEt3ABA-r1-84:-KSfJNRT5hNpoWwQLutw:b-896fix) ([raw file](https://github.com/mozilla/telemetry-airflow/blob/6960b1c989730258610a345e1cc41693a4a46661/ansible/files/spark/airflow.sh#L84)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Please use lower case variables names
</details>
Done.

---

*[ansible/files/spark/airflow.sh, line 93 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-airflow/45#-KSKoMvhXN7q7MEt3ABA-r1-93:-KSfJRujxCacXnyunzt5:b-896fix) ([raw file](https://github.com/mozilla/telemetry-airflow/blob/6960b1c989730258610a345e1cc41693a4a46661/ansible/files/spark/airflow.sh#L93)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> What about using `rc` and do the assignment in each branch?
</details>
Done.

---

*[ansible/files/spark/airflow.sh, line 124 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-airflow/45#-KSKoMvhXN7q7MEt3ABA-r1-124:-KSfJSvHXVWZEh1yxQdC:b-896fix) ([raw file](https://github.com/mozilla/telemetry-airflow/blob/6960b1c989730258610a345e1cc41693a4a46661/ansible/files/spark/airflow.sh#L124)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> If we use only `rc` we can remove this part.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-airflow/45)*

**Comment by vitillo** [comment]
Has this been tested?

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-airflow/45#-:-KSfmAuFu7XbzSHSpl7d:b-r4g367)*

**Comment by maurodoglio** [comment]
I tested it on a fresh cluster, both the success and the fail case.

**Comment by vitillo** [comment]
Looks good, thanks!

**Comment by vitillo** [comment]
I have deployed the new script.

**Pull request merged**